src/Indicator.vala

```diff
@@ -87,13 +87,15 @@ public class DateTime.Indicator : Wingpanel.Indicator {
     private void idle_update_events () {
         if (update_events_idle_source > 0) {
             Source.remove (update_events_idle_source);
-            update_events_idle_source = 0;
         }
+
         update_events_idle_source = Idle.add (update_events);
     }
 
     private bool update_events () {
-        if (event_grid != null) event_grid.destroy();
+        if (event_grid != null) {
+            event_grid.destroy();
```

Comment:
Sorry - I missed the lack of a space before the bracket here :-(

Sorry to be fussy.